jax/scipy/linalg.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import scipy.linalg
+
+from .. import lax_linalg
+from ..numpy.lax_numpy import _wraps
+from ..numpy import lax_numpy as np
+
+
+@_wraps(scipy.linalg.cholesky)
+def cholesky(a, lower=False, overwrite_a=False, check_finite=True):
+  del overwrite_a, check_finite
+  if not lower:
+    raise NotImplementedError(
+        "The lower=False case of Cholesky is not implemented.")
+
+  return lax_linalg.cholesky(a)
+
+
+@_wraps(scipy.linalg.qr)
+def qr(a, overwrite_a=False, lwork=None, mode="full", pivoting=False,
+       check_finite=True):
+  del overwrite_a, lwork, check_finite
+  if pivoting:
+    raise NotImplementedError(
+        "The pivoting=True case of qr is not implemented.")
+  if mode in ("full", "r"):
+    full_matrices = True
+  elif mode == "economic":
+    full_matrices = True
+  else:
+    raise ValueError("Unsupported QR decomposition mode '{}'".format(mode))
+  q, r = lax_linalg.qr(a, full_matrices)
+  if mode == "r":
+    return r
+  return q, r
+
+
+@_wraps(scipy.linalg.solve_triangular)
+def solve_triangular(a, b, trans=0, lower=False, unit_diagonal=False,
+                     overwrite_b=False, debug=None, check_finite=True):
+  del overwrite_b, debug, check_finite
```

Comment:
We might want to warn if these flags are set to anything but their default values (by which I mean, set in any way that doesn't match the code that follows).